framework/sessions/__init__.py

```diff
@@ -171,5 +172,14 @@ def before_request():
 def after_request(response):
     if session.data.get('auth_user_id'):
         session.save()
+        #import here to prevent circular import error
+        from framework.auth.core import _get_current_user
 
+        user = _get_current_user()
+        if user:
+            try:
+                user.date_last_login = datetime.utcnow()
+                user.save()
+            except:
```

Comment:
Hm, then maybe it would make sense `date_last_login`  in a `before_request` hook, before any work is done in the view?